tests/rados/customer_scenarios.py

```diff
@@ -0,0 +1,385 @@
+    """rados put and get in a parallel task"""
+    with parallel() as p:
+        p.spawn(do_rados_put, client_node, pool_name, 40)
+        p.spawn(do_rados_get, client_node, pool_name, 40)
+    mon_db_size_list.append(get_mondb_size(mon_nodes[0], mon_daemons))
+
+    # deleting a previously created pool to increase OSD operations and map changes
+    pname = "delete_pool"
+    detete_pool(node=node, pool=pname)
+
+    # Proceeding to reboot 1 OSD from each host to trigger rebalance & Backfill
+    for val in osd_nodes:
+        daemon_info = run_ceph_command(node, action="daemon_info")
+        osd_daemons = [entry for entry in daemon_info if entry["daemon_type"] == "osd"]
+        for osd in osd_daemons:
+            if re.search(osd["hostname"], val.hostname):
+                cmd = f"podman restart {osd['container_id']}"
+                log.info(
+                    f"rebooting osd-{osd['daemon_id']} on host {osd['hostname']}. Command {cmd}"
+                )
+                val.exec_command(sudo=True, cmd=cmd)
+                break
+
+    # Re-weighting the OSd's based on usage to trigger rebalance
+    # todo: Verify re-balancing process on OSD's ( PG movement across cluster)
+    # todo: Add re-balancing based on crush item provided
+    # BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1766702
+    reweight_crush_items(node=node)
+
+    """
+    Waiting for 2 hours for cluster to get to active + clean state.
+    Rationale: during cluster activities like backfill, rebalance, osd map change cause Mon DB to be updated.
+    Hence, we can wait till mon DB updates are completed, after which DB size should be reduced,
+    by trimming the old mappings once the new mappings are added.
+    """
+    end_time = datetime.datetime.now() + datetime.timedelta(seconds=7200)
+    while end_time > datetime.datetime.now():
+        ceph_health_status = run_ceph_command(node, action="ceph_report")["health"]
+        recovery_tuple = ("OSD_DOWN", "PG_AVAILABILITY", "PG_DEGRADED", "SLOW_OPS")
+        daemon_info = run_ceph_command(node, action="daemon_info")
+        mon_daemons = [entry for entry in daemon_info if entry["daemon_type"] == "mon"]
+        mon_db_size_list.append(get_mondb_size(mon_nodes[0], mon_daemons))
+        if not any(
+            key in ceph_health_status["checks"].keys() for key in recovery_tuple
+        ):
+            log.info(
+                f"The recovery and back-filling of the OSD is completed"
+                f"Sleeping for {mon_db_trim_wait_dur} Seconds after clean for mon to remove old mappings"
+            )
+            time.sleep(mon_db_trim_wait_dur / 2)
+            mon_db_size_list.append(get_mondb_size(mon_nodes[0], mon_daemons))
+            time.sleep(mon_db_trim_wait_dur)
+            break
+        log.info(
+            f"Waiting for active + clean. Active aletrs: {ceph_health_status['checks'].keys()},"
+            f" checking status again in 1 minute"
+        )
+        time.sleep(60)
```

Comment:
As mentioned in earlier comment, Once we are sure that slow ops are generated, we can fix this. If you see, there is no failure if there are no slow ops. Also we are just waiting here on count. Need to improve the method to track if the same operation is stuck for long time or there are lots of operations generated/ cleared in succession, such that there is always at least 1 operation running.

will include this in todo.

Reply:
In my testing ( did not have more objects/load ), still tuning the configs, the test was completed within 10 min, so only around 10 entries. 

I can remove the logging part if logging is what we are concerned, but I felt it would be good as it would give us the current warns on cluster every minute.

felt would be helpful if needs debugging in case of failures